core/templates/dev/head/domain/topic/TopicObjectFactory.ts

```diff
@@ -90,6 +109,10 @@ angular.module('oppia').factory('TopicObjectFactory', [
         issues.push('Topic name should not be empty.');
       }
 
+      if (this._thumbnailDataUrl === '') {
```

Comment:
What about for `_abbreviatedTopicName`?
Also, isn't this just for publishing? Or is this function only checked before publishing?

Reply:
Added validaiton.